sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -2225,6 +2246,154 @@ object Radar {
         })
     }
 
+    /**
+     * Sends a custom event.
+     *
+     * @see [](https://radar.io/documentation/api#sen-event)
+     *
+     * @param[name] The name of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun sendEvent(name: String, metadata: JSONObject?, callback: RadarSendEventCallback) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        trackOnce(object : RadarTrackCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback.onComplete(status)
+                    }
+
+                    return
+                }
+
+                apiClient.sendEvent(name, metadata, user, object : RadarApiClient.RadarSendEventApiCallback {
+                    override fun onComplete(status: RadarStatus, res: JSONObject?, event: RadarEvent?) {
+                        if (status != RadarStatus.SUCCESS) {
+                            handler.post {
+                                callback.onComplete(status)
+                            }
+
+                            return
+                        }
+
+                        val finalEvents: MutableList<RadarEvent>
```

Comment:
Nick had a comment on iOS that it's a bit funky to do this in the top-level `Radar` file, makes more sense to live int he API client layer, especially since it's repeated across both `apiClient.sendEvent` callbacks, fixed it in this commit - https://github.com/radarlabs/radar-sdk-ios/pull/190/commits/ffb90bdbedaef0451aeb0854c31b9a6a463813a1

Reply:
I refactored these two blocks of code into a new function in the `RadarAPIClient`.